LogicalTypes.md

```diff
@@ -709,13 +709,18 @@ to values. `MAP` must annotate a 3-level structure:
 
 * The outer-most level must be a group annotated with `MAP` that contains a
   single field named `key_value`. The repetition of this level must be either
-  `optional` or `required` and determines whether the list is nullable.
+  `optional` or `required` and determines whether the map is nullable.
 * The middle level, named `key_value`, must be a repeated group with a `key`
-  field for map keys and, optionally, a `value` field for map values.
+  field for map keys and, optionally, a `value` field for map values. It must
+  not contain any other values.
 * The `key` field encodes the map's key type. This field must have
-  repetition `required` and must always be present.
+  repetition `required` and must always be present. I must be placed at the 0th
+  position of the `key_value` group. It is suggested to use a primitive as the
+  type.
 * The `value` field encodes the map's value type and repetition. This field can
-  be `required`, `optional`, or omitted.
+  be `required`, `optional`, or omitted. It must be placed at the 1st position
+  of the `key_value` group if present. In case of not present, the map can be
+  either represented with all null values or as a set of keys.
 
```

Comment:
I just realized that the value can be emitted 🤔 When would a case be like that?

Reply:
Yeah, I've never seen an actual use-case implemented for this one either. I am open to require having a value but since it is out there for a while now, I am not sure if we can do it. Maybe a suggestion to always specify a value and let it be an optional with all nulls if one would not use it?

Reply:
Should we move omitted value field to the backward compatibility section, which is all about cases that are not suggested?

Reply:
I'm not sure backward compatibility section would be the best place for this. That section is mainly for supporting Parquet files that were written before the proper spec.
Anyway, it seems similar to the nested key question. @JFinis, WDYT?

Reply:
> I just realized that the value can be emitted 🤔 When would a case be like that?

Perhaps when using a `MAP` to encode a set? 🤷

Reply:
Unfortunately, parquet-cpp does not implement this: https://github.com/apache/arrow/blob/c3601a97a0718ae47726e6c134cbed4b98bd1a36/cpp/src/parquet/arrow/schema.cc#L578-L583

Reply:
@wgtmac, @etseidl, WDYT, shall I simply remove this last sentence about handling the case of "missing" values? I don't think we can change the original spec because it is clearly stated that value can be omitted. Then, it is up to the implementation how they support/not support this case.

Reply:
The original [discussion](https://github.com/apache/parquet-format/pull/17/commits/3135c61ac07f67154f8c2eb5eac16789257eeb6f#r19687863) backs up this interpretation. I think it's good to leave the explanation in.